pcdsdevices/lxe.py

```diff
@@ -414,6 +414,170 @@ def format_status_info(self, status_info):
+    """
+
+    tab_component_names = True
+
+    verbose_name = 'Laser X-ray Timing'
+
+    _tgt_time = Cpt(EpicsSignal, ':PHASCTL:DELAY_SET', auto_monitor=True,
+                    kind='omitted',
+                    doc='The internal nanosecond-expecting signal.'
+                    )
+    setpoint = Cpt(_ScaledUnitConversionDerivedSignal,
+                   derived_from='_tgt_time',
+                   derived_units='s',
+                   original_units='ns',
+                   kind='hinted',
+                   doc='Setpoint which handles the timing conversion.',
+                   limits=(-100e-6, 100e-6),
+                   )
+    notepad_setpoint = Cpt(NotepadLinkedSignal, ':lxt:OphydSetpoint',
+                           notepad_metadata={'record': 'ao',
+                                             'default_value': 0.0},
+                           kind='omitted'
+                           )
+    notepad_readback = Cpt(NotepadLinkedSignal, ':lxt:OphydReadback',
+                           notepad_metadata={'record': 'ao',
+                                             'default_value': 0.0},
+                           kind='omitted'
+                           )
+    user_offset = Cpt(NotepadLinkedSignal, ':lxt:OphydOffset',
+                      notepad_metadata={'record': 'ao', 'default_value': 0.0},
+                      kind='normal',
+                      doc='A Python-level user offset.'
+                      )
+
+    # The phase shifter will be moved after the above record is touched, so
+    # use its done status:
+    done = Cpt(EpicsSignal, ':PHASCTL:DELAY_MOVING', auto_monitor=True, kind='omitted')
+    done_value = 0
+
+    def __init__(self, prefix='', *, egu=None, **kwargs):
+        if egu not in (None, 's'):
+            raise ValueError(
+                f'{self.__class__.__name__} is pre-configured to work in units'
+                f' of seconds.'
+            )
+        super().__init__(prefix, egu='s', **kwargs)
+
+    @user_offset.sub_value
+    def _offset_changed(self, value, **kwargs):
+        """
+        The user offset was changed.  Update the setpoint attribute.
+        """
+        self.setpoint.user_offset = value
+
+    def _setup_move(self, position):
+        """Update the notepad setpoint, move, and do not wait."""
+        try:
+            signal = self.notepad_setpoint
+            if signal.connected and signal.write_access:
+                if signal.get(use_monitor=True) != position:
+                    signal.put(position, wait=False)
+        except Exception as ex:
+            self.log.debug('Failed to update notepad setpoint to position %s',
+                           position, exc_info=ex)
+        super()._setup_move(position)
+
+        # Something is wrong with done signal, just sleep and pretend
+        time.sleep(1)
+        self._move_changed(value=1-self.done_value)
+        self._move_changed(value=self.done_value)
```

Comment:
I notice this suspicious bit is copied from the old class. Is this behavior still needed with the new setups? Could this be responsible for some of the strange behavior you're seeing?

Reply:
Ah, perhaps. I will try removing that old behavior when I can.

Reply:
Yeah, I think all three lines here should be removed if the moving/done signal value is accurate.